esp-hal/src/lib.rs

```diff
@@ -215,17 +215,34 @@ pub enum Cpu {
 
 /// Which core the application is currently executing on
 pub fn get_core() -> Cpu {
-    Cpu::from_repr(get_raw_core()).unwrap()
+    // This works for both RISCV and Xtensa because both
+    // get_raw_core functions return zero, _or_ something
+    // greater than zero; 1 in the case of RISCV and 0x2000
+    // in the case of Xtensa.
+    match get_raw_core() {
+        0 => Cpu::ProCpu,
+        #[cfg(all(multicore, riscv))]
+        1 => Cpu::AppCpu,
+        #[cfg(all(multicore, xtensa))]
```

Comment:
```suggestion
        #[cfg(all(multi_core, riscv))]
```

Reply:
@ProfFan I'm guessing this might be a copy-paste bug, but please keep this branch `xtensa` specific :)